vividus-plugin-applitools/src/main/java/org/vividus/visual/eyes/converter/ExamplesTableToApplitoolsVisualChecksConverter.java

```diff
@@ -0,0 +1,213 @@
+import org.vividus.visual.eyes.model.ApplitoolsVisualCheck;
+import org.vividus.visual.model.VisualActionType;
+import org.vividus.visual.screenshot.BaselineIndexer;
+
+public class ExamplesTableToApplitoolsVisualChecksConverter extends
+        AbstractParameterConverter<ExamplesTable, List<ApplitoolsVisualCheck>> implements ApplitoolsVisualCheckFactory
+{
+    private static final String BASELINE_NAME_FIELD = "baselineName";
+    private static final String ACTION_FIELD = "action";
+    private static final String EXECUTE_API_KEY_FIELD = "executeApiKey";
+    private static final String READ_API_KEY_FIELD = "readApiKey";
+    private static final String HOST_APP_FIELD = "hostApp";
+    private static final String HOST_OS_FIELD = "hostOS";
+    private static final String VIEWPORT_SIZE_FIELD = "viewportSize";
+    private static final String MATCH_LEVEL_FIELD = "matchLevel";
+    private static final String SERVER_URI_FIELD = "serverUri";
+    private static final String APP_NAME_FIELD = "appName";
+    private static final String BATCH_NAME_FIELD = "batchName";
+    private static final String BASELINE_ENV_NAME_FIELD = "baselineEnvName";
+    private static final String ELEMENTS_TO_IGNORE_FIELD = "elementsToIgnore";
+    private static final String AREAS_TO_IGNORE_FIELD = "areasToIgnore";
+
+    private static final List<String> SUPPORTED_FIELDS = List.of(BASELINE_NAME_FIELD, ACTION_FIELD,
+            EXECUTE_API_KEY_FIELD, READ_API_KEY_FIELD, HOST_APP_FIELD, HOST_OS_FIELD, VIEWPORT_SIZE_FIELD,
+            MATCH_LEVEL_FIELD, SERVER_URI_FIELD, APP_NAME_FIELD, BATCH_NAME_FIELD, BASELINE_ENV_NAME_FIELD,
+            ELEMENTS_TO_IGNORE_FIELD, AREAS_TO_IGNORE_FIELD);
+
+    private String executeApiKey;
+    private String readApiKey;
+    private String hostApp;
+    private String hostOS;
+    private RectangleSize viewportSize;
+    private MatchLevel matchLevel;
+    private URI serverUri;
+    private String appName = "Application";
+    private String baselineEnvName;
+
+    private final Map<String, BatchInfo> batchStorage = new ConcurrentHashMap<>();
+
+    private final ScreenshotParametersFactory<ScreenshotConfiguration> screenshotParametersFactory;
+    private final BaselineIndexer baselineIndexer;
+
+    public ExamplesTableToApplitoolsVisualChecksConverter(
+            ScreenshotParametersFactory<ScreenshotConfiguration> screenshotParametersFactory,
+            BaselineIndexer baselineIndexer)
+    {
+        this.screenshotParametersFactory = screenshotParametersFactory;
+        this.baselineIndexer = baselineIndexer;
+    }
+
+    @Override
+    public List<ApplitoolsVisualCheck> convertValue(ExamplesTable table, Type type)
+    {
+        List<String> unsupportedFields = table.getHeaders().stream().filter(field -> !SUPPORTED_FIELDS.contains(field))
+                .collect(Collectors.toList());
+        Validate.isTrue(unsupportedFields.isEmpty(), "Unknown visual check parameters: %s",
```

Comment:
```suggestion
        Validate.isTrue(unsupportedFields.isEmpty(), "Unknown Applitools configuration parameters: %s",
```
we need to be consistent in naming (including both docs and code)